aiohasupervisor/models/mounts.py

```diff
@@ -0,0 +1,133 @@
+"""Models for Supervisor mounts."""
+
+from abc import ABC
+from dataclasses import dataclass, field
+from enum import StrEnum
+from pathlib import PurePath
+from typing import Literal
+
+from .base import Request, ResponseData
+
+# --- ENUMS ----
+
+
+class MountType(StrEnum):
+    """MountType type."""
+
+    CIFS = "cifs"
+    NFS = "nfs"
+
+
+class MountUsage(StrEnum):
+    """MountUsage type."""
+
+    BACKUP = "backup"
+    MEDIA = "media"
+    SHARE = "share"
+
+
+class MountState(StrEnum):
+    """MountState type."""
+
+    ACTIVE = "active"
+    ACTIVATING = "activating"
+    DEACTIVATING = "deactivating"
+    FAILED = "failed"
+    INACTIVE = "inactive"
+    MAINTENANCE = "maintenance"
+    RELOADING = "reloading"
+
+
+class MountCifsVersion(StrEnum):
+    """Mount CIFS version."""
+
+    LEGACY_1_0 = "1.0"
+    LEGACY_2_0 = "2.0"
+
+
+# --- OBJECTS ----
+
+
+@dataclass(frozen=True)
+class Mount(ABC):
+    """Mount ABC type."""
+
+    usage: MountUsage
+    server: str
+    port: int | None = field(kw_only=True, default=None)
+
+
+@dataclass(frozen=True)
+class CIFSMount(ABC):
+    """CIFSMount ABC type."""
+
+    share: str
+    version: MountCifsVersion | None = field(kw_only=True, default=None)
+
+
+@dataclass(frozen=True)
+class NFSMount(ABC):
+    """NFSMount ABC type."""
+
+    path: PurePath
+
+
+@dataclass(frozen=True)
+class MountResponse(ABC):
+    """MountResponse model."""
+
+    name: str
+    read_only: bool
+    state: MountState | None
+
+
+@dataclass(frozen=True)
+class MountRequest(ABC):  # noqa: B024
```

Comment:
Why do we need to disable the ruff rule here but not in the other ABC classes?

Reply:
I believe its because the only field it has is optional. As long as there is at least one required field that doesn't show up.